### DISCUSSION TITLE
Offer a way to check when the navigator is ready

### DISCUSSION THREAD

**Discussion by zackify** [expo/router#508] [Ideas]
Edit: here is a solution in v2:

```tsx
 let state = useRootNavigationState();

if (!state.index && state.index !== 0) return null;

return <Redirect to="/" />
```

If state.index isn't set yet, you cannot call router.replace or return redirect components. If you return early for that part, you can navigate the moment the app opens and initializes.

I want to display a modal when a push notification comes in.

When the app is closed all the way, and opens up. The notification listener fires before navigation is ready.

For now this is what I have to do:

```ts
import { getNavigationContainerRef } from "expo-router/src/NavigationContainer";

...
let ref = getNavigationContainerRef();

while (ref.isReady() === false) {
  await new Promise((resolve) => setTimeout(resolve, 100));
}
// now i can router.push inside of my noficiations event listener
router.push()
```

This is the only way i can ensure the app doesn't break, by importing the navigation container from the app, which seems like something that may break in a future version.

**Comment by wcastand** [comment]
I would like that as well, for now in the background task that fire on the notification when the app was killed, i save the screen into AsyncStorage and in the app i look if a value is set and if so i navigate to it.

**Reply by zackify** [reply]
Updated my comment, found a way in v2, but again, this feels like something that will break again in the next version haha
